### PR TITLE
CSS-8009 Add cos-lite integration

### PR DESCRIPTION
**Pull request by AmberCharitos**
This PR adds integrated observability to the Trino charm in line with DOG2 requirements. It adds:

- Loki, prometheus and grafana libraries & requirer/provider interfaces
- Grafana dashboard template for monitoring Trino metrics 
- required properties added to the trino config file.

Please note that this PR is dependent on [this PR](https://github.com/canonical/charmed-trino-rock/pull/4) and integration tests will fail until the rock PR has been merged.

Screenshot from 2024-04-05 09-31-09
Screenshot from 2024-04-04 17-14-43

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,6 +114,58 @@ juju relate ranger-k8s postgresql-k8s
...
+juju offer grafana:grafana-dashboard
+
+# Relate trino to the cos-lite apps:
+juju relate trino-k8s admin/cos.grafana
+juju relate trino-k8s admin/cos.loki
+juju relate trino-k8s admin/cos.prometheus
+```
+
+After relating the trino server charm to cos-lite services,
+we need, for the time being, to attach the promtail-bin resource so that
```
Comment (gtato) [inline README.md:143]: can't this be part of the rock?

**Review by gtato — APPROVED** [review]
lg

File: README.md (hunk 1 above)
Comment (AmberCharitos) [reply README.md:143]: This is a good point.. making necessary updates..

File: README.md [hunk 2]
```diff
@@ -114,6 +114,38 @@ juju relate ranger-k8s postgresql-k8s
...
+juju relate trino-k8s admin/cos.grafana
+juju relate trino-k8s admin/cos.loki
+juju relate trino-k8s admin/cos.prometheus
+```
+
+```bash
+# Access grafana with username "admin" and password:
+juju run grafana/0 -m cos get-admin-password --wait 1m
+# Grafana is listening on port 3000 of the app ip address.
+# Dashboard can be accessed under "Trino Server Metrics", make sure to select the juju model which contains your Tino charm.
```
Comment (kelkawi-a) [inline README.md:146]: Trino* charm

**Review by kelkawi-a — APPROVED** [review]
LGTM, great work!

**Pull request merged**